sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/batch/BulkExecutor.java

```diff
@@ -820,6 +844,12 @@ private Mono<CosmosBatchResponse> executeBatchRequest(PartitionKeyRangeServerBat
         options.setExcludedRegions(cosmosBulkExecutionOptions.getExcludedRegions());
         options.setKeywordIdentifiers(cosmosBulkExecutionOptions.getKeywordIdentifiers());
 
+        CosmosEndToEndOperationLatencyPolicyConfig e2eLatencyPolicySnapshot =
```

Comment:
just setting the e2e timeout policy here may not be enough, as we did not wire up the timeout handling down in the RxDocumentClientImpl

Reply:
Fixed